android/src/main/java/com/reactnativestripesdk/Mappers.kt

```diff
@@ -419,34 +419,39 @@ internal fun mapToAddress(addressMap: ReadableMap?, cardAddress: Address?): Addr
 }
 
 internal fun mapToBillingDetails(billingDetails: ReadableMap?, cardAddress: Address?): PaymentMethod.BillingDetails? {
-  if (billingDetails == null) {
-    return null
+  if (billingDetails == null && cardAddress == null) {
+    return null;
   }
   val address = Address.Builder()
-    .setPostalCode(getValOr(billingDetails, "addressPostalCode"))
-    .setCity(getValOr(billingDetails, "addressCity"))
-    .setCountry(getValOr(billingDetails, "addressCountry"))
-    .setLine1(getValOr(billingDetails, "addressLine1"))
-    .setLine2(getValOr(billingDetails, "addressLine2"))
-    .setState(getValOr(billingDetails, "addressState"))
-
-    cardAddress?.let { ca ->
-      ca.postalCode?.let {
-        if (it.isNotEmpty()) {
-          address.setPostalCode(it)
-        }
-      }
-      ca.country?.let {
-        address.setCountry(it)
+  if (billingDetails != null) {
+    address
+      .setPostalCode(getValOr(billingDetails, "addressPostalCode"))
+      .setCity(getValOr(billingDetails, "addressCity"))
+      .setCountry(getValOr(billingDetails, "addressCountry"))
+      .setLine1(getValOr(billingDetails, "addressLine1"))
+      .setLine2(getValOr(billingDetails, "addressLine2"))
+      .setState(getValOr(billingDetails, "addressState"))
+  }
+  cardAddress?.let { ca ->
+    ca.postalCode?.let {
+      if (it.isNotEmpty()) {
+        address.setPostalCode(it)
       }
     }
+    ca.country?.let {
+      address.setCountry(it)
+    }
+  }
```

Comment:
Nit: I know this is what was here before, but maybe we should change it to be easier to read, like:

```
  if (cardAddress?.postalCode?.isNotEmpty() == true) {
    addressBuilder.setPostalCode(cardAddress.postalCode)
  }
  if (cardAddress?.country?.isNotEmpty() == true) {
    addressBuilder.setCountry(cardAddress.country)
  }
```

?

Reply:
Yup, I agreed! this is much more readable. 

Perhaps we should drop the `==true` as well?

Reply:
Ah, `isNotEmpty()` returns Boolean or null, so `==true` is necessary.